otherlibs/stdune/src/path.ml

```diff
@@ -862,6 +862,12 @@ let external_of_in_source_tree x = external_of_local x ~root:(Lazy.force abs_roo
 
 let reach t ~from =
   match t, from with
+  | External t, External from ->
+    (match
+       String.drop_prefix ~prefix:(External.to_string from ^ "/") (External.to_string t)
```

Comment:
Why is this needed? Can't you pass external paths as absolute paths?

Reply:
It's for `Odoc_new.local_path_of_findlib_path` - there's a comment explaining what it does. Is there a better way to do that?

Reply:
We have `Path.drop_prefix` which might be more sensible. It also gives a `Local.Path.t` which I believe is what you actually want. I'm still a bit confused about the `Odoc_new.local_path_of_findlib_path` function however. It seems everywhere you use it, you split on `/` and then take the head, so I think there is some other property you are trying to extract here.

Reply:
eurgh - Path.drop_prefix is not always doing what I want!

```
# obj_dir = '/Users/jon/devel/odoc/_opam/lib/ocaml-compiler-libs/shadow'
# local = '-compiler-libs/shadow'
```

This is because there are two findlib paths - `/Users/jon/devel/odoc/_opam/lib` and `/Users/jon/devel/odoc/_opam/lib/ocaml` - and it's the second is definitely a string prefix of the lib above, but not a _path_ prefix.

Reply:
I don't - you're spot on, I really want the path prefix, and getting back a `Path.Local.t` would be great. Sorry, I wasn't clear above. The above is what `Path.drop_prefix` is giving me, not what I actually want - ie, I think `Path.drop_prefix` is broken. For the above, what I'd like is to return the local path representing `ocaml-compiler-libs/shadow`.

Reply:
OK, let me try to repro in a unit test.

Reply:
Yep, that's definitely a bug. This is what happens when unit tests are not written! :)

Reply:
Or actually maybe intended. I guess what we really need is a `Path.drop_path_prefix` function which doesn't exist.

Reply:
I've created a test https://github.com/ocaml/dune/pull/8870 so that we can discuss this behaviour there.